var/spack/repos/builtin/packages/sundials/package.py

```diff
@@ -27,138 +27,341 @@
 import sys
 
 
-class Sundials(Package):
+class Sundials(CMakePackage):
     """SUNDIALS (SUite of Nonlinear and DIfferential/ALgebraic equation
     Solvers)"""
 
-    homepage = "http://computation.llnl.gov/casc/sundials/"
-    url = "http://computation.llnl.gov/projects/sundials-suite-nonlinear-differential-algebraic-equation-solvers/download/sundials-2.6.2.tar.gz"
+    homepage = "https://computation.llnl.gov/projects/sundials"
+    url = "https://computation.llnl.gov/projects/sundials/download/sundials-2.7.0.tar.gz"
+    maintainers = ['cswoodward', 'gardner48']
 
-    version('2.7.0', 'c304631b9bc82877d7b0e9f4d4fd94d3')
+    # ==========================================================================
+    # Versions
+    # ==========================================================================
+
+    version('3.0.0-beta-2',
+            git='https://github.com/LLNL/sundials.git', tag='v3.0.0-beta-2')
+    version('2.7.0', 'c304631b9bc82877d7b0e9f4d4fd94d3', preferred=True)
     version('2.6.2', '3deeb0ede9f514184c6bd83ecab77d95')
 
+    # ==========================================================================
+    # Variants
+    # ==========================================================================
+
+    # SUNDIALS solvers
+    sun_solvers = ['CVODE', 'CVODES', 'ARKODE', 'IDA', 'IDAS', 'KINSOL']
+
+    for pkg in sun_solvers:
+        variant(pkg, default=True,
+                description='Enable %s solver' % pkg)
+
+    # Real type
+    variant(
+        'precision',
+        default='double',
+        description='''real type precision''',
+        values=('single', 'double', 'extended'),
+        multi=False
+    )
+
+    # Index type
+    variant(
+        'indextype',
+        default='int64_t',
+        description='''index integer type''',
+        values=('int32_t', 'int64_t'),
```

Comment:
to be consistent with other packages (i.e. at least `petsc`, `mump`, `superlu-dist`, `hypre`), could you please make it into a single bool variant:
```
int64 [off]          True, False       Compile with 64bit indices
```

Reply:
I've made this a single bool option but does the default need to be False? Prior to the current beta release `sundials` only supported 64bit indices.

Reply:
> Prior to the current beta release sundials only supported 64bit indices.

i would say we shall keep it as `True` then, so that default values are working, i.e. `spack install sundials` should build.

And after you have a next stable release we can switch it to `False` by default to be consistent with defaults in other packages.